File: output/csbc_semish_2026/x_ai_grok_code_fast_1/zero/python/Fita Colorida.py

import sys

input = sys.stdin.read
data = input().split()

N = int(data[0])
lista = list(map(int, data[1:]))

dist = [float('inf')] * N

# Primeira passada: esquerda para direita
last_zero = -1
for i in range(N):
    if lista[i] == 0:
        dist[i] = 0
        last_zero = i
    else:
        if last_zero != -1:
            dist[i] = i - last_zero

# Segunda passada: direita para esquerda
last_zero = N
for i in range(N-1, -1, -1):
    if lista[i] == 0:
        last_zero = i
    else:
        if last_zero != N:
            dist[i] = min(dist[i], last_zero - i)

# Output
for i in range(N):
    if lista[i] == 0:
        print(0, end=' ')
    else:
        print(min(dist[i], 9), end=' ')
print()

File: output/csbc_semish_2026/x_ai_grok_code_fast_1/zero/cpp/Fita Colorida.cpp

#include <bits/stdc++.h>

using namespace std;

int main() {
    int N;
    cin >> N;
    vector<int> A(N);
    for (int i = 0; i < N; i++) cin >> A[i];
    vector<int> dist(N, 10000);
    for (int i = 0; i < N; i++) {
        if (A[i] == 0) dist[i] = 0;
    }
    // primeira passagem
    for (int i = 1; i < N; i++) {
        dist[i] = min(dist[i], dist[i - 1] + 1);
    }
    // segunda passagem
    for (int i = N - 2; i >= 0; i--) {
        dist[i] = min(dist[i], dist[i + 1] + 1);
    }
    // output
    for (int i = 0; i < N; i++) {
        if (A[i] == 0) {
            cout << 0;
        } else {
            cout << min(dist[i], 9);
        }
        if (i < N - 1) cout << " ";
        else cout << endl;
    }
    return 0;
}
